libs/community/langchain_community/vectorstores/vdms.py

```diff
@@ -0,0 +1,1630 @@
+DEFAULT_VDMS_CONNECTION = {"host": "localhost", "port": 55555}
+INVALID_DOC_METADATA_KEYS = ["_distance", "content", "blob"]
+INVALID_METADATA_VALUE = ["Missing property", None, {}]  # type: List
+logger = logging.getLogger(__name__)
+
+
+def _len_check_if_sized(x: Any, y: Any, x_name: str, y_name: str) -> None:
+    """
+    Check that sizes of two variables are the same
+
+    Args:
+        x: Variable to compare
+        y: Variable to compare
+        x_name: Name for variable x
+        y_name: Name for variable y
+    """
+    if isinstance(x, Sized) and isinstance(y, Sized) and len(x) != len(y):
+        raise ValueError(
+            f"{x_name} and {y_name} expected to be equal length but "
+            f"len({x_name})={len(x)} and len({y_name})={len(y)}"
+        )
+    return
+
+
+class VDMS(VectorStore):
+    """Wrapper around Intel Lab's VDMS for vector-store workloads.
+
+    To use, you should have both:
+    - the ``vdms`` python package installed
+    - a host (str) and port (int) associated with a deployed VDMS Server
+
+    Visit https://github.com/IntelLabs/vdms/wiki more information.
+
+    IT IS HIGHLY SUGGESTED TO NORMALIZE YOUR DATA.
+
+    Args:
+        collection_name: Name of data collection [Default: langchain]
+        distance_strategy: Method used to calculate distances. VDMS supports
+            "L2" (euclidean distance) or "IP" (inner product) [Default: L2]
+        engine: Underlying implementation for indexing and computing distances.
+            VDMS supports TileDBDense, TileDBSparse, FaissFlat, FaissIVFFlat,
+            and Flinng [Default: FaissFlat]
+        connection_args: Dictionary defining "host" and "port" for VDMS server
+        embedding_function: Any embedding function implementing
+            `langchain_core.embeddings.Embeddings` interface.
+        relevance_score_fn: Function for obtaining relevance score
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.vectorstores import VDMS
+            from langchain_community.embeddings.huggingface import HuggingFaceEmbeddings
+
+            vectorstore = VDMS(
+                collection_name="langchain-demo",
+                distance_strategy="L2",
+                engine="FaissFlat"
+                connection_args={"host": "localhost", "port": 55555},
+                embedding_function=HuggingFaceEmbeddings(),
+            )
+    """
+
+    def __init__(
+        self,
+        embedding_function: Optional[Embeddings] = None,
+        collection_name: str = DEFAULT_COLLECTION_NAME,  # DescriptorSet name
+        distance_strategy: Union[DistanceMetric, str] = DEFAULT_DISTANCE_METRIC,
+        engine: Union[IndexEngine, str] = DEFAULT_SEARCH_ENGINE,
+        connection_args: Dict[str, Any] = DEFAULT_VDMS_CONNECTION,
+        relevance_score_fn: Optional[Callable[[float], float]] = None,
+    ) -> None:
+        # Connect to VDMS Server
+        self._create_connection_alias(connection_args)
```

Comment:
This forces a network request on every instantiation of the vectorstore -- this will make user code slow in many production settings -- the code should ideally either memoize the client or else allow a user to pass their own client

Reply:
I modified the code and the client is passed to the vector store. Thanks!